tasks/ldap/src/main/java/com/walmartlabs/concord/plugins/ldap/LdapTaskCommon.java

```diff
@@ -44,6 +44,8 @@
     private static final int MAX_RETRIES = 3;
     private static final long RETRY_DELAY = 3000;
 
+    private LdapContext activeConnection = null;
```

Comment:
Why we need this as a field?

Reply:
Only reason is because the connection context object gets recreated on a retry.
https://github.com/walmartlabs/concord-plugins/blob/462442e2a26199b9918c7ce4c650fe4b632bd9ef/tasks/ldap/src/main/java/com/walmartlabs/concord/plugins/ldap/LdapTaskCommon.java#L199

And, at least historically, we were trying to make sure it was closed--but that probably wouldn't work on a retry
https://github.com/walmartlabs/concord-plugins/blob/f578e895f09a4e37a1f96f35c27e1741b346fd6c/tasks/ldap/src/main/java/com/walmartlabs/concord/plugins/ldap/LdapTaskCommon.java#L202-L206

Reply:
I think the connection handling code should be withing `withRetry`.
Or make `activeConnection` a thread-local variable.

Reply:
looks like there are still connection leak:
`establishConnection` creates connection on each retry and we close only latest connection...
we can try something like this:
```
            // create SearchRequest
            return withRetry(MAX_RETRIES, RETRY_DELAY, () -> {
                LdapContext connection = null;
                try {
                    connection = establishConnection(cfg);
                    return connection.search(searchBase, searchFilter, searchControls);
                } finally {
                    if (connection != null) {
                        try {
                            connection.close();
                        } catch (NamingException e) {
                            // ignore
                        }
                    }
                }
            });       
```

Reply:
See next commit for fix